pkg/controller/common/helpers_test.go

```diff
@@ -53,6 +57,18 @@ func TestValidateIgnition(t *testing.T) {
 	isValid2 = ValidateIgnition(testIgn3Config)
 	require.Nil(t, isValid2)
 
+	// Test that file modes do not have special bits (sticky, setuid, setgid) set
+	// https://bugzilla.redhat.com/show_bug.cgi?id=2038240
+	invalidMode := 0o1777
+
+	testIgn3Config.Storage.Files[0].Mode = &invalidMode
+	isValid2 = ValidateIgnition(testIgn3Config)
```

Comment:
Now that you've factored out the mode checking function, would it be better to make a unit test to target just those functions? Or if not, just so I can learn, why would it be better to leave as is?

Reply:
Those functions are already called via validateingnition and zack's existing test cases already test valid mode, so he only has to check invalid mode.

Reply:
I mean why is he calling `ValidateIgnition` rather than `validateIgn3FileModes`

Reply:
it is called via validate ignition and not necessary to call it specifically.